server/monitor/src/main/java/org/apache/accumulo/monitor/Monitor.java

```diff
@@ -428,10 +428,11 @@ private GCStatus fetchGcStatus() {
       var path = ServiceLock.path(context.getZooKeeperRoot() + Constants.ZGC_LOCK);
       List<String> locks = ServiceLock.validateAndSort(path, zk.getChildren(path.toString()));
       if (locks != null && !locks.isEmpty()) {
-        Optional<ServiceLockData> sld =
-            ServiceLockData.parse(zk.getData(path + "/" + locks.get(0)));
-        if (sld.isPresent()) {
-          address = sld.orElseThrow().getAddress(ThriftService.GC);
+        address = ServiceLockData.parse(zk.getData(path + "/" + locks.get(0)))
+            .map(sld -> sld.getAddress(ThriftService.GC)).orElse(null);
+        if (address == null) {
+          log.warn("Unable to contact the garbage collector (no address)");
+          return null;
```

Comment:
The rest of the code is ok with this returning null?

Before these change could this code have thrown an NPE?

Not sure about the log level of warn for the new log message.  Sometimes the GC is down and that is expected and the warn would not be useful in that case.  When the GC is down and its not expected, these log messages are probably not a great way to do detect that.  I would go with debug for this log message.

Reply:
Before this change, this would have resulted in attempting to make a connection to a server using null for the address. That would have resulted in some exception, though I don't know off the top of my head whether that would be an NPE or an IOException, and caught further down, which would have resulted in a similar message, but reporting that it failed to connect to the GC at the "null" address. This just short-circuits that attempt to connect using a null address and fail later with a non-sensical message. It's not strictly necessary, but I thought it'd be a better behavior.